src/primitive.rs

```diff
@@ -83,11 +119,41 @@ impl TryFrom<f64> for FiniteF64 {
     }
 }
 
+impl TryFrom<i64> for FiniteF64 {
+    type Error = TemporalError;
+    fn try_from(value: i64) -> Result<Self, Self::Error> {
+        let result = f64::from_i64(value)
+            .ok_or(TemporalError::range().with_message("number exceeded a valid range."))?;
+        Ok(Self(result))
+    }
+}
+
+impl TryFrom<u64> for FiniteF64 {
+    type Error = TemporalError;
+    fn try_from(value: u64) -> Result<Self, Self::Error> {
+        let result = f64::from_u64(value)
+            .ok_or(TemporalError::range().with_message("number exceeded a valid range."))?;
+        Ok(Self(result))
+    }
+}
+
 impl TryFrom<i128> for FiniteF64 {
     type Error = TemporalError;
     fn try_from(value: i128) -> Result<Self, Self::Error> {
         let result = f64::from_i128(value)
-            .ok_or(TemporalError::range().with_message("days exceeded a valid range."))?;
+            .ok_or(TemporalError::range().with_message("number exceeded a valid range."))?;
+        if !result.is_finite() {
+            return Err(TemporalError::range().with_message("number value is not a finite value."));
+        }
```

Comment:
Is there any case where `f64::from_i128` would result in a non-finite f64? I guess same question for `u128` too.

Reply:
Good question. Not entirely sure where there is or not, but we could definitely test it. Just to be safe though. This check is mostly a sanity check. We could `debug_assert` instead. But we should probably have tests around that then.